lightningd/peer_control.c

```diff
@@ -1482,14 +1490,29 @@ static int peer_channel_announced(struct peer *peer, const u8 *msg)
 	if (!fromwire_hsmctl_node_announcement_sig_reply(msg, NULL, &sig))
 		fatal("HSM returned an invalid node_announcement sig");
 
-	/* We got the signature for out provisional node_announcement back
-	 * from the HSM, create the real announcement and forward it to
-	 * gossipd so it can take care of forwarding it. */
-	announcement = create_node_announcement(tmpctx, ld, &sig);
+	/* Manually create wire-ready signature */
+	sig_wire = tal_arr(tmpctx, u8, 0);
```

Comment:
I don't like pulling the serialization out of the methods that are dedicated to it. I'd rather pass in a timestamp explicitly that makes sure we sign with the same timestamp, than having outside manipulation of the packet.